include/secp256k1_musig.h

```diff
@@ -7,348 +7,151 @@
 extern "C" {
 #endif
 
-#include <stdint.h>
+#include <stddef.h>
 
-/** This module implements a Schnorr-based multi-signature scheme called MuSig
- * (https://eprint.iacr.org/2018/068.pdf). It is compatible with bip-schnorr.
+/** This module implements a Schnorr-based multi-signature scheme called MuSig2
+ * (https://eprint.iacr.org/2020/1261). It is compatible with BIP-340 ("Schnorr").
  * There's an example C source file in the module's directory
  * (src/modules/musig/example.c) that demonstrates how it can be used.
  *
- * The documentation in this include file is for reference and may not be sufficient
- * for users to begin using the library. A full description of API usage can be found
- * in src/modules/musig/musig.md
+ * The module also supports BIP-341 ("Taproot") public key tweaking and adaptor
+ * signatures as described in
+ * https://github.com/ElementsProject/scriptless-scripts/pull/24
+ *
+ * It is recommended to read the documentation in this include file carefully.
+ * Further notes on API usage can be found in src/modules/musig/musig.md
+ *
+ * You will sometimes read on the internet that MuSig2 uses "two nonces". This
+ * is not wrong, but only technical detail we don't want to bother the reader
+ * with. Therefore, we only refer to the singular "nonce".
```

Comment:
```suggestion
 * You may know that the MuSig2 scheme uses two "nonces" instead of one. This
 * is not wrong, but only a technical detail we don't want to bother the user
 * with. Therefore, the API only uses to the singular term "nonce".
```